### DISCUSSION TITLE
Turn off activity information in logs

### DISCUSSION THREAD

**Discussion by lostmsu** [dotnet/runtime#102135] [Q&A] [answered]
How do I turn off this thing? https://github.com/dotnet/runtime/issues/34305

> => SpanId: xxx, TraceId:yyy, ParentId: zzz => my stuff

I am not big enough for distributed logging, this is just pure pollution.

Relatedly, IMHO, this should not be enabled by default, as by default this data only makes sense for ASP.NET and in my app there's nothing that would set these IDs in a way that has meaning until I manually add it anyway.

**Comment by En3Tho** [comment, marked as answer]
There is a sample in the linked issue. Use `ActivityTrackingOptions.None`.

**Reply by lostmsu** [reply]
Is there a way to do it from appsettings?

**Reply by En3Tho** [reply]
I'm not sure. It's weird that you have this information reported by default. I believe simple call to `AddLogging()` for example does not do that for you. Maybe there is something else that is using activity/telemetry?

**Reply by lostmsu** [reply]
I am using

```csharp
var builder = Host.CreateApplicationBuilder(args);
builder.Logging.ClearProviders();
builder.Logging.AddConsole();
builder.Logging.AddFile(...);
builder.Logging.AddApplicationInsights(...);
var host = builder.Build();
var logs = host.Services.GetRequiredService<ILoggerFactory>();
```

Then upon request I do
```csharp
var log = logs.CreateLogger<RequestHandler>();
log.BeginScope("Request {ID}", id);

```

**Reply by En3Tho** [reply]
I think `builder.Logging.AddApplicationInsights(...);` line can add this behavior. ApplicationInsights sounds like it's Azure telemetry thing if I'm not wrong.
